src/Krank/Checkers/Ignore.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Krank.Checkers.Ignore ( IgnoreCommand(..)
+                             , filterViolations
+                             ) where
+
+import PyF (fmt)
+
+import qualified Data.ByteString.Char8 as ByteString
+import Data.ByteString.Char8 (ByteString)
+import Data.HashMap.Strict as HashM
+import qualified Data.List as DataL
+import qualified Text.Regex.PCRE.Heavy as RE
+
+import Krank.Types
+
+data IgnoreCommand = IgnoreLine deriving (Show, Eq)
+
+-- | This regex represents a krank ignore marker
+ignoreRe :: RE.Regex
+-- NOTE: \b at the beginning is really import for performances
+-- because it dramatically reduces the number of backtracks
```

Comment:
Technically here that's not true. It reduces the number of backtrack in the url parser because the url parser starts by an optional string and then matchs an unspecified number of arbitrary characters.

This means that the regex engine must start by trying the optional string. If it succeed, it will then match any unspecified number of arbitrary characters. If by any bad luck the match fails, it will come back the the beginning and try without the optional string and do the same process. And this is done for any offset in the input string.

By forcing the first char of the match to be something specific (i.e. a word boundary `\b`), the regex engine first start by looking for a word boundary, then it does the complex analysis with possible backtracking. The change here is important, because instead of applying the costly process on all offset of the input string, it only applies it on all offsets which match the specific start pattern (i.e. a word boundary). The ratio of word boundaries over characters in a text is easily 5 (I assume that it may be the mean of the length of words in a standard piece of program.